snapcraft/storeapi/v2/_api_schema.py

```diff
@@ -0,0 +1,221 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Attributes that the jsonschema from the Snap Store originally require but
+# Snapcraft does not have been commented out from this schema originally
+# imported from
+# https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-channel-map
+CHANNEL_MAP_JSONSCHEMA = {
+    "additionalProperties": False,
+    "properties": {
+        "channel-map": {
+            "items": {
+                "additionalProperties": False,
+                "properties": {
+                    "architecture": {"type": "string"},
+                    "channel": {
+                        "description": 'The channel name, including "latest/" for the latest track.',
+                        "type": "string",
+                    },
+                    "expiration-date": {
+                        "description": "The date when this release expires, in RFC 3339 format. If null, the release does not expire.",
+                        "format": "date-time",
+                        "type": ["string", "null"],
+                    },
+                    "progressive": {
+                        "additionalProperties": False,
+                        "properties": {
+                            "key": {"type": ["string", "null"]},
+                            "paused": {"type": ["boolean", "null"]},
+                            "percentage": {"type": ["number", "null"]},
+                        },
+                        "required": ["key", "paused", "percentage"],
+                        "type": "object",
+                    },
+                    "revision": {"type": "integer"},
+                    "when": {
+                        "description": "The date when this release was made, in RFC 3339 format.",
+                        "format": "date-time",
+                        "type": "string",
+                    },
+                },
+                "required": [
+                    "architecture",
+                    "channel",
+                    "expiration-date",
+                    "progressive",
+                    "revision",
+                    # "when"
```

Comment:
I assume fields were commented out that we don't care about?  I think it's fair to leave them, even if we don't care about them (yet).

Reply:
yes, exactly so, I added a comment mentioning this at the top.